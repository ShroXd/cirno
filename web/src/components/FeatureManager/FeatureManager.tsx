import { Switch } from '~/components/ui/switch'
import { useFeatures } from '~/hooks/feature/useFeatures'

export const FeatureManager = () => {
  const { getAllFeatures, toggleFeature } = useFeatures()

  return (
    <>
      {getAllFeatures().map(feature => (
        <div className='flex items-center justify-between' key={feature.id}>
          <Switch
            id={feature.id}
            checked={feature.enabled}
            onCheckedChange={() => toggleFeature(feature.id)}
          />
        </div>
      ))}
    </>
  )
}
