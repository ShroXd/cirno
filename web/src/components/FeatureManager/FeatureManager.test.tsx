import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { describe, expect, it } from 'vitest'

import { FeatureManager } from './FeatureManager'
import { renderWithContext } from '~/test/test-utils'

describe('FeatureManager', () => {
  it('should render without a11y violations', async () => {
    const { container } = render(<FeatureManager />)
    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render correctly', () => {
    const { container } = renderWithContext(<FeatureManager />)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })
})
