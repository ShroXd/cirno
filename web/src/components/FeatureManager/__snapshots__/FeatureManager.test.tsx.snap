// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeatureManager > should render correctly 1`] = `
<div
  class="flex items-center justify-between"
>
  <div
    class="inline-flex items-center"
  >
    <div
      class="relative inline-block w-8 h-4 cursor-pointer rounded-full"
    >
      <input
        class="peer appearance-none w-8 h-4 absolute bg-blue-gray-100 rounded-full cursor-pointer transition-colors duration-300 checked:bg-gray-900 peer-checked:border-gray-900 peer-checked:before:bg-gray-900"
        id=":r0:"
        type="checkbox"
      />
      <label
        class="bg-white w-5 h-5 border border-blue-gray-100 rounded-full shadow-md absolute top-2/4 -left-1 -translate-y-2/4 peer-checked:translate-x-full transition-all duration-300 cursor-pointer before:content[''] before:block before:bg-blue-gray-500 before:w-10 before:h-10 before:rounded-full before:absolute before:top-2/4 before:left-2/4 before:-translate-y-2/4 before:-translate-x-2/4 before:transition-opacity before:opacity-0 hover:before:opacity-10 peer-checked:border-gray-900 peer-checked:before:bg-gray-900"
        for=":r0:"
      />
    </div>
    <label
      class="text-gray-700 font-light select-none cursor-pointer mt-px ml-3 mb-0"
      for=":r0:"
    >
      <div>
        <p
          class="block antialiased font-sans text-base leading-relaxed text-blue-gray-900 font-medium"
        >
          feature.edgeCaseTrigger.name
        </p>
        <p
          class="block antialiased font-sans text-sm leading-normal text-gray-700 font-normal"
        >
          feature.edgeCaseTrigger.description
        </p>
      </div>
    </label>
  </div>
</div>
`;
